src/gui/toolbarMenubar/model/ColorPalette.cpp

```diff
@@ -0,0 +1,154 @@
+}
+
+
+NamedColor Palette::getColorAt(size_t i) const {
+    if (i >= namedColors.size()) {
+        i = i % namedColors.size();
+        g_warning("There are more Coloritems in the Toolbar then your Palette defines.\n"
+                  "Hence, cycling through palette from the beginning.");
+    }
+
+    return namedColors.at(i);
+}
+
+NamedColor* Palette::getColorPointerAt(size_t i) {
+    if (i >= namedColors.size()) {
+        i = i % namedColors.size();
+        g_warning("There are more Coloritems in the Toolbar then your Palette defines.\n"
+                  "Hence, cycling through palette from the beginning.");
+    }
+
+    return &namedColors.at(i);
+}
+
+size_t Palette::size() const { return namedColors.size(); }
+
+
+void Palette::create_default(fs::path filepath) {
+    std::ofstream myfile{filepath};
+    myfile << "GIMP Palette"
+           << "\n";
+    myfile << "Name: Xournal Default Palette"
+           << "\n";
+    myfile << "#"
+           << "\n";
+    myfile << 0 << " " << 0 << " " << 0 << " "
+           << "Black"
+           << "\n";
+    myfile << 0 << " " << 128 << " " << 0 << " "
+           << "Green"
+           << "\n";
+    myfile << 0 << " " << 192 << " " << 255 << " "
+           << "Light Blue"
+           << "\n";
+    myfile << 0 << " " << 255 << " " << 0 << " "
+           << "Light Green"
+           << "\n";
+    myfile << 51 << " " << 51 << " " << 204 << " "
+           << "Blue"
+           << "\n";
+    myfile << 128 << " " << 128 << " " << 128 << " "
+           << "Gray"
+           << "\n";
+    myfile << 255 << " " << 0 << " " << 0 << " "
+           << "Red"
+           << "\n";
+    myfile << 255 << " " << 0 << " " << 255 << " "
+           << "Magenta"
+           << "\n";
+    myfile << 255 << " " << 128 << " " << 0 << " "
+           << "Orange"
+           << "\n";
+    myfile << 255 << " " << 255 << " " << 0 << " "
+           << "Yellow"
+           << "\n";
+    myfile << 255 << " " << 255 << " " << 255 << " "
+           << "White"
+           << "\n";
+}
+
+std::istream& operator>>(std::istream& str, Header& header) {
+    // inspired by
+    // https://codereview.stackexchange.com/questions/38879/parsing-text-file-in-c
+    std::string line;
+    Header tmp;
+    if (std::getline(str, line)) {
+        std::istringstream iss{line};
+        if (getline(iss, tmp.attribute, ':') && getline(iss, tmp.value, '\n')) {
+            tmp.attribute = StringUtils::trim(tmp.attribute);
+            tmp.value = StringUtils::trim(tmp.value);
+
+            // parsing was successful hence results can be stored permanently
+            header.swap(tmp);
+        } else {
+            // One operation failed.
+            // So set the state on the main stream
+            // to indicate failure.
+            str.setstate(std::ios::failbit);
+        }
+    }
+    return str;
+}
+
+void Header::swap(Header& other) {
+    std::swap(attribute, other.attribute);
+    std::swap(value, other.value);
+}
```

Comment:
If that function is only used to swap in the input stream function, it can be removed.

Reply:
Will be done :heavy_check_mark: .